lib/rubocop/cop/performance/string_bytesize.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # Checks for calls to `#bytes` counting method and suggests using `bytesize` instead.
+      # The `bytesize` method is more efficient and directly returns the size in bytes,
+      # avoiding the intermediate array allocation that `bytes.size` incurs.
+      #
+      # @safety
+      #   This cop is unsafe because it assumes that the receiver
+      #   responds to `#bytesize` method.
+      #
+      # @example
+      #   # bad
+      #   string_var.bytes.count
+      #   "foobar".bytes.size
+      #
+      #   # good
+      #   string_var.bytesize
+      #   "foobar".bytesize
+      class StringBytesize < Base
+        extend AutoCorrector
+
+        MSG = 'Use `String#bytesize` instead of calculating the size of the bytes array.'
+        RESTRICT_ON_SEND = %i[size length count].freeze
+
+        def_node_matcher :string_bytes_method?, <<~MATCHER
+          (call (call !nil? :bytes) {:size :length :count})
```

Comment:
Can you add a test for `bytes.size` with no receiver?

Reply:
There's already a test for this case: https://github.com/rubocop/rubocop-performance/pull/474/files#diff-6bd9463058468a2a8ab5cd49bd6023946d72b460a0382226c7d16649a5db1421R78-R82

Reply:
My bad, missed it. Thanks!